plugins/com.google.cloud.tools.eclipse.jdt.launching/tests/com/google/cloud/tools/eclipse/jdt/launching/MultipleConnectionsTest.java

```diff
@@ -0,0 +1,174 @@
+		connector = new SocketListenMultiConnector();
+		Map<String, Connector.Argument> defaults = connector.getDefaultArguments();
+		assertTrue(defaults.containsKey("acceptCount"));
+		assertEquals(1, ((Connector.IntegerArgument) defaults.get("acceptCount")).intValue());
+	}
+
+	/**
+	 * Ensure out-of-the-box settings mimics previous behaviour of accepting a
+	 * single connection
+	 * 
+	 * @throws IOException
+	 */
+	@Test
+	public void testDefaultBehaviour() throws CoreException {
+		connector = new SocketListenMultiConnector();
+		Map<String, String> arguments = new HashMap<>();
+		arguments.put("port", Integer.toString(port));
+		arguments.put("acceptCount", "1");
+		connector.connect(arguments, new NullProgressMonitor(), launch);
+		Thread.yield();
+
+		assertTrue("first connect should succeed", connect());
+		assertFalse("second connect should fail", connect());
+	}
+
+	/**
+	 * Ensure connector accepts a single connection
+	 * 
+	 * @throws InterruptedException
+	 */
+	@Test
+	public void testSingleConnectionBehaviour() throws CoreException, InterruptedException {
+		connector = new SocketListenMultiConnector();
+		Map<String, String> arguments = new HashMap<>();
+		arguments.put("port", Integer.toString(port));
+		arguments.put("acceptCount", "1");
+		connector.connect(arguments, new NullProgressMonitor(), launch);
+		Thread.sleep(200);
+
+		assertTrue("first connect should succeed", connect());
+		assertFalse("second connect should fail", connect());
+	}
+
+	/**
+	 * Ensure out-of-the-box settings mimics previous behaviour of accepting a
+	 * single connection
+	 * 
+	 * @throws InterruptedException
+	 */
+	@Test
+	public void testTwoConnectionsBehaviour() throws CoreException, InterruptedException {
+		connector = new SocketListenMultiConnector();
+		Map<String, String> arguments = new HashMap<>();
+		arguments.put("port", Integer.toString(port));
+		arguments.put("acceptCount", "2");
+		connector.connect(arguments, new NullProgressMonitor(), launch);
+		Thread.sleep(200);
+
+		assertTrue("first connect should succeed", connect());
+		assertTrue("second connect should succeed", connect());
+	}
+
+	/**
+	 * Ensure out-of-the-box settings mimics previous behaviour of accepting a
+	 * single connection
+	 * 
+	 * @throws InterruptedException
+	 */
+	@Test
+	public void testUnlimitedConnectionsBehaviour() throws CoreException, InterruptedException {
+		connector = new SocketListenMultiConnector();
+		Map<String, String> arguments = new HashMap<>();
+		arguments.put("port", Integer.toString(port));
+		arguments.put("acceptCount", "0");
+		connector.connect(arguments, new NullProgressMonitor(), launch);
+		Thread.sleep(200);
+
+		for (int i = 0; i < 10; i++) {
+			assertTrue("connection " + i + " should succeed", connect());
+		}
+	}
+
+	@After
+	public void tearDown() throws DebugException {
+		launch.terminate();
+	}
+
+	private boolean connect() {
+		boolean result = true;
+		try (Socket s = new Socket()) {
+			try {
```

Comment:
To differentiate between `IOException`s from the _try-with_'s socket-closure and `IOException`s from connecting and writing on the socket, indicating a problem with the remote.